src/services/rfqm_service.ts

```diff
@@ -439,11 +443,60 @@ export class RfqmService {
     /**
      * Process an orderHash as an RfqmJob. Throws an error if job must be retried
      */
-    // tslint:disable: prefer-function-over-method
-    public async processRfqmJobAsync(orderHash: string): Promise<void> {
-        logger.info({ orderHash }, 'finished processing job');
-        // tslint:disable-next-line: custom-no-magic-numbers
-        return Math.random() > 0.5 ? Promise.reject('failing') : Promise.resolve();
+    public async processRfqmJobAsync(orderHash: string, workerAddress: string): Promise<void> {
+        logger.info({ orderHash }, 'start processing job');
+
+        // Get job
+        const job = await this.dbUtils.findJobByOrderHashAsync(orderHash);
+        if (job === undefined) {
+            throw new NotFoundError(`job for orderHash ${orderHash} not found`);
+        }
+
+        // Basic validation
+        await this._validateJobAsync(orderHash, job);
+        const { calldata, makerUri, order, fee } = job;
+
+        // Start processing
+        await this.dbUtils.updateRfqmJobAsync(orderHash, {
+            status: RfqmJobStatus.Processing,
+        });
+
+        // Validate w/Eth Call
+        try {
+            await this._blockchainUtils.decodeMetaTransactionCallDataAndValidateAsync(calldata!, workerAddress);
+        } catch (e) {
+            // Terminate with an error transition
+            await this.dbUtils.updateRfqmJobAsync(orderHash, {
+                status: RfqmJobStatus.Failed,
+                statusReason: 'eth_call failed',
+            });
+            return;
+        }
+
+        // Get last look from MM
+        const submitRequest: SubmitRequest = {
+            order: storedOrderToRfqmOrder(order!),
+            orderHash,
+            fee: storedFeeToFee(fee!),
+        };
+
+        const shouldProceed = await this._quoteServerClient.confirmLastLookAsync(makerUri!, submitRequest);
```

Comment:
more logging here would be good too, we'll want to keep track of the ratio of confirmed quotes to rejected

Reply:
very true! Will be able to get this from the DB as well, but being able to alert on this could be very useful too